design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -0,0 +1,211 @@
+# Customer Orders API
+
+## Overview
+
+The GraphQL API should provide a possibility to retrieve orders, shipments, invoices, credit memos for the logged in customer. The current schema allows fetching only simple order details and doesn't provide a possibility to fetch order details by a number.
+
+The proposed solution is deprecation of `customerOrders` query and `orders` field to the `customer` query:
+
+```graphql
+# query to return a list of all customer orders
+type Customer {
+    orders (
+        filter: CustomerOrdersFilterInput
+        currentPage: Int = 1 # current page of the customer order list. default is 1.
+        pageSize: Int = 20 # page size for the customer orders list. default is 20.
+    ): CustomerOrders
+}
+
+input CustomerOrdersFilterInput {
+    order_number: String!
+}
+
+# collection of customer orders.
+type CustomerOrders {
+    items: [CustomerOrder]! # collection of customer orders that contains individual order details.
+    page_info: SearchResultPageInfo
+    total_count: Int
+}
+```
+
+> Right now, we don't introduce the filter input for such entities like invoice, credit memo, shipment as all these entities are related to the same order and GraphQL resolvers anyway at first resolve the order and after that other child entities. But, in future, such filter input can be introduced as optional argument.
+
+## Order Type Schema
+
+The proposed type for the customer order might look like:
+
+```graphql
+# order details
+type CustomerOrder {
+    order_date: String! # date when the order was placed.
```

Comment:
Please convert all field comments to '@doc' directives.

Reply:
Does it make sense to do for the proposal? It's not a schema file.